### PR TITLE
Explicitly mark fields as private via MBEDTLS_PRIVATE(...)

### PR DESCRIPTION
**Pull request by hanno-arm**
The fact that those fields were not explicitly marked private currently is an oversight due to concurrent merging of PRs during the run-up to the release of Mbed TLS 3.0.

### CODE REVIEW

File: ChangeLog.d/session_export_private.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Changes
+    * Mark mbedtls_ssl_session.exported as private. This was an oversight
+      during the run-up to the release of Mbed TLS 3.0. The field was
+      never intended to be marked public.
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/session_export_private.txt:4]: I'd rather not create the impression that not-explicitly-private fields are public. In principle, fields are private by default ([“are not considered API compatibility breaks: (…) Removing a field from a structure, unless the field is documented as public.”](https://github.com/ARMmbed/mbedtls/blob/development/BRANCHES.md#backwards-compatibility)).
```suggestion
    * Explicitly mark mbedtls_ssl_session.exported as private. This was an
      oversight during the run-up to the release of Mbed TLS 3.0. The field
      was never intended to be public.
```

Comment (hanno-arm) [reply ChangeLog.d/session_export_private.txt]: I assume this means that this PR shouldn't be marked as an API break then - @gilles-peskine-arm @daverodgman

Comment (hanno-arm) [reply ChangeLog.d/session_export_private.txt]: Followed your suggestion, @gilles-peskine-arm

**Review by gilles-peskine-arm — DISMISSED** [review]

**Review by daverodgman — APPROVED** [review]
Seems consistent with our non-API-break guidelines:

"Here are some examples of changes that are common in minor releases of Mbed TLS, and are not considered API compatibility breaks: (…)
* Removing a field from a structure, unless the field is documented as public."

which is effectively what's happening here.

**Review by gilles-peskine-arm — APPROVED** [review]

**Comment by hanno-arm** [comment]
Ping @daverodgman

**Review by daverodgman — APPROVED** [review]

**Pull request merged**